crates/core/src/diff/fragment/mod.rs

```diff
@@ -316,12 +323,16 @@ impl TryFrom<FragmentDiff> for Fragment {
     type Error = MergeError;
     fn try_from(value: FragmentDiff) -> Result<Self, MergeError> {
         match value {
-            FragmentDiff::UpdateRegular { children } => {
+            FragmentDiff::UpdateRegular { children, statics } => {
                 let mut new_children : HashMap<String, Child> = HashMap::new();
                 for (key, cdiff) in children.into_iter() {
                     new_children.insert(key, cdiff.try_into()?);
                 }
-                let statics = Statics::Statics(vec!["".into(); new_children.len()]);
+                let statics = if let Some(statics) = statics {
+                    statics
+                } else {
+                        Statics::Statics(vec!["".into(); new_children.len()])
```

Comment:
It's not important but the indentation looks a bit too much here. 

```suggestion
                    Statics::Statics(vec!["".into(); new_children.len()])
```